lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -343,15 +350,33 @@ bool SslClientConnection::connectSocket() {
 
   ERR_clear_error();
 
-  int ret = SSL_connect(_ssl);
+  int ret = -1;
+  int errorDetail = -1;
+
+  if (_isTelemetrics) {
+    auto start = steady_clock::now();
+    while ((ret = SSL_connect(_ssl)) == -1) {
+      errorDetail = SSL_get_error(_ssl, ret);
+      if (_isInterrupted) {
+        break;
+      }
+      auto end = steady_clock::now();
+      if ((errorDetail != SSL_ERROR_WANT_READ &&
+           errorDetail != SSL_ERROR_WANT_WRITE) ||
+          duration_cast<seconds>(end - start).count() >= _connectTimeout) {
+        break;
+      }
+    }
+  } else {
+    ret = SSL_connect(_ssl);
+    errorDetail = SSL_get_error(_ssl, ret);
```

Comment:
I guess we should not call `SSL_get_error` unconditionally here. What if no error has happened at all? Is it then still ok to call `SSL_get_error` here?